### PR TITLE
Properly set hasClass

### PR DESCRIPTION
**Pull request by mattmoor**
This was changed in #1277

/hold

I wanna verify in serving first

### CODE REVIEW

File: codegen/cmd/injection-gen/generators/packages.go [hunk 1]
```diff
@@ -193,13 +193,13 @@ func extractCommentTags(t *types.Type) map[string]map[string]string {
...
-	if !has {
-		return
+func extractReconcilerClassTag(tags map[string]map[string]string) (string, bool) {
+	vals, ok := tags["genreconciler"]
+	if !ok {
+		return "", false
 	}
-	classname, _ = vals["class"]
-	return
+	classname, has := vals["class"]
```
Comment (whaught) [inline codegen/cmd/injection-gen/generators/packages.go:201]: is `return vals["class"]` the same here?

Comment (mattmoor) [reply codegen/cmd/injection-gen/generators/packages.go:201]: No, I tried that first :(

Go special cases subscript on maps to have one or two values, but it isn't smart enough to use the latter here.

**Comment by mattmoor** [comment]
/hold cancel

This works for me

**Pull request merged**